### PR TITLE
br: make br support FIPS

### LINKED ISSUE #48419: make br support FIPS
## Feature Request
ref #47948

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #48419 

Problem Summary:

### What is changed and how it works?
imitate the #47949 
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```
make build_br_fips
go tool nm bin/br |grep boring
 1f55d60 t local.crypto/internal/boring._Cfunc_EVP_AEAD_CTX_open_wrapper.abi0
 1f55f40 t local.crypto/internal/boring._Cfunc_EVP_AEAD_CTX_seal_wrapper.abi0
 1f56120 t local.crypto/internal/boring._Cfunc__goboringcrypto_AES_cbc_encrypt.abi0
 1f56280 t local.crypto/internal/boring._Cfunc__goboringcrypto_AES_ctr128_encrypt.abi0
 1f563e0 t local.crypto/internal/boring._Cfunc__goboringcrypto_AES_decrypt.abi0
 1f564a0 t local.crypto/internal/boring._Cfunc__goboringcrypto_AES_encrypt.abi0
 1f56560 t local.crypto/internal/boring._Cfunc__goboringcrypto_AES_set_decrypt_key.abi0
 1f56640 t local.crypto/internal/boring._Cfunc__goboringcrypto_AES_set_encrypt_key.abi0
 1f56720 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_bin2bn.abi0
 1f56800 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_bn2bin_padded.abi0
 1f568e0 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_bn2le_padded.abi0
 1f68580 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_free
 1f569c0 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_free.abi0
 1f56a40 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_le2bn.abi0
 1f56b20 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_new.abi0
 1f56b80 t local.crypto/internal/boring._Cfunc__goboringcrypto_BN_num_bytes.abi0
 1f56c00 t local.crypto/internal/boring._Cfunc__goboringcrypto_BORINGSSL_bcm_power_on_self_test.abi0
 1f56c60 t local.crypto/internal/boring._Cfunc__goboringcrypto_ECDSA_sign.abi0
 1f56dc0 t local.crypto/internal/boring._Cfunc__goboringcrypto_ECDSA_size.abi0
 1f56e40 t local.crypto/internal/boring._Cfunc__goboringcrypto_ECDSA_verify.abi0
```

```
./bin/br -V
Release Version: v7.6.0-alpha-203-gb3fbb4951d-dirty-fips
Git Commit Hash: b3fbb4951dd16b9e22987ef57845c83dc33da99a
Git Branch: br_fips
Go Version: go1.21.0 X:boringcrypto
 ...

./bin/tidb-lightning -V           
Release Version: v7.6.0-alpha-203-gb3fbb4951d-dirty-fips
Git Commit Hash: b3fbb4951dd16b9e22987ef57845c83dc33da99a
Git Branch: br_fips
Go Version: go1.21.0 X:boringcrypto
...

./bin/tidb-lightning-ctl -V 
Release Version: v7.6.0-alpha-203-gb3fbb4951d-dirty-fips
Git Commit Hash: b3fbb4951dd16b9e22987ef57845c83dc33da99a
Git Branch: br_fips
Go Version: go1.21.0 X:boringcrypto
...

./bin/tidb-server -V       
Release Version: v7.6.0-alpha-203-gb3fbb4951d-dirty-fips
Edition: Community
Git Commit Hash: b3fbb4951dd16b9e22987ef57845c83dc33da99a
Git Branch: br_fips
UTC Build Time: 2023-11-16 05:34:40
GoVersion: go1.21.0 X:boringcrypto
...
```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by overvenus** [comment]
> ./bin/tidb-server -V
>  Release Version: v7.6.0-alpha-883-g40d25a807a-fips-fips
>  Git Commit Hash: 40d25a807abdb6c30c2f46290eef4440343581c9
>  Git Branch: br_fips
>  Go Version: go1.21.0 X:boringcrypto
>  ...
> 

`v7.6.0-alpha-883-g40d25a807a-fips-fips` the two fips looks strange.

File: Makefile [hunk 1]
```diff
@@ -312,7 +312,10 @@ br_web:
 	@cd br/web && npm install && npm run build
 
 build_br:
-	CGO_ENABLED=1 $(GOBUILD) $(RACE_FLAG) -ldflags '$(LDFLAGS) $(CHECK_FLAG)' -o $(BR_BIN) br/cmd/br/*.go
+	CGO_ENABLED=1 $(GOBUILD) $(RACE_FLAG) -ldflags '$(LDFLAGS) $(CHECK_FLAG)' -o $(BR_BIN) ./br/cmd/br
+
+build_br_fips:
+	GOEXPERIMENT=boringcrypto CGO_ENABLED=1 $(GOBUILD) -tags boringcrypto $(RACE_FLAG) -ldflags '$(LDFLAGS) $(CHECK_FLAG)' -o $(BR_BIN) ./br/cmd/br
```
Comment (overvenus) [inline Makefile:318]: No need build_br_fips, try `ENABLE_FIPS=1`

**Review by overvenus — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
> br/OWNERS

You can /approve your own PR. I have tried some days ago @Leavrth

**Pull request merged**